x/mongo/driver/topology/server.go

```diff
@@ -290,6 +292,32 @@ func (s *Server) ProcessHandshakeError(err error, startingGenerationNumber uint6
 		return
 	}
 
+	isTimeout := func(err error) bool {
+		for err != nil {
+			if netErr, ok := err.(net.Error); ok && netErr.Timeout() {
+				return true
+			}
+			// Handle the case where an error has been replaced by "net.errCanceled", which isn't
+			// exported and can't be compared directly. In this case, just compare the error message.
+			if err.Error() == "operation was canceled" {
+				return true
+			}
+			if wrapper, ok := err.(interface{ Unwrap() error }); ok {
```

Comment:
Super-nit: not sure if this is still the case, but IIRC we try to avoid if/else blocks when possible. Instead of using `break`, you can have this conditional `continue` and then `return false` outside of it. The existing `return false` outside the loop can probably be removed at that point.

Reply:
The compiler actually complains about a missing return if I remove the `return false` after the loop. Will remove the if/else and replace it with a
```go
if !ok {
    break
}
```